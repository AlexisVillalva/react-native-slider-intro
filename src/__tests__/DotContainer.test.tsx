import React from 'react';
import renderer from 'react-test-renderer';
import DotContainer from '../components/DotContainer';
import { Animated } from 'react-native';

test('renders correctly', () => {
  const tree = renderer
    .create(
      <DotContainer
        arrayOfSlideIndex={[1, 2, 3]}
        fixDotBackgroundColor={'grey'}
        fixDotOpacity={0.35}
        dotWidth={12}
        _moveSlideDotX={new Animated.Value(0)}
        _moveSlideDotMarginX={new Animated.Value(0)}
        animatedDotBackgroundColor={'white'}
        navContainerMaxSize={0.3}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
