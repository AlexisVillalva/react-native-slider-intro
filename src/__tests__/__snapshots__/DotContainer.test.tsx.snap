// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
      Object {
        "maxWidth": 0.3,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "flexDirection": "row",
        },
        Object {
          "maxWidth": 0.3,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "borderRadius": 100,
              "flex": 1,
            },
            Object {
              "backgroundColor": "grey",
              "height": 12,
              "maxHeight": 12,
              "maxWidth": 12,
              "opacity": 0.35,
              "width": 12,
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "borderRadius": 100,
              "flex": 1,
            },
            Object {
              "backgroundColor": "grey",
              "height": 12,
              "maxHeight": 12,
              "maxWidth": 12,
              "opacity": 0.35,
              "width": 12,
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "borderRadius": 100,
              "flex": 1,
            },
            Object {
              "backgroundColor": "grey",
              "height": 12,
              "maxHeight": 12,
              "maxWidth": 12,
              "opacity": 0.35,
              "width": 12,
            },
          ]
        }
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "flexDirection": "row",
          "position": "absolute",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "flex-start",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 100,
            "flex": 1,
            "height": 12,
            "marginLeft": 0,
            "maxHeight": 12,
            "maxWidth": 0,
            "width": 0,
          }
        }
      />
    </View>
  </View>
</View>
`;
